sssom/io.py

```diff
@@ -23,6 +23,8 @@ def convert_file(input_path: str, output_path: str = None, output_format: str =
     Returns:
 
     """
+    if not os.path.exists(os.path.dirname(output_path)):
```

Comment:
It is not necessary for output path to be anything, because the correct behaviour here should be: if legal_filepath(output_path): write fo file, else: write to stdout.. Not sure whether we did this for convert, but this should be the consistent behaviour for all methods.. So if we want this new and improved condition plus error message, we need to first make sure that output path is actually something other than None or `"-"`.

Reply:
Okay, but this is not obvious from any of the documentation or the code itself.

I made a fix in 3c8e046 to check that it's a pathlike before doing this check

Reply:
Would `-` pass as a "path-like object"?

Reply:
No, but this sort of points out that the entire way this is being handled is broken. Only the CLI should be worrying about parsing that stuff. There's a "smart open" function, which is also a huge anti-pattern. I will send an alternate PR to show how this should be done instead

Reply:
Ok, thank you!

Reply:
See #145